### ISSUE TITLE
cdk bootstrap can't handle state REVIEW_IN_PROGRESS

### ISSUE THREAD

**Issue by chizou** [aws/aws-cdk#19623]
### What is the problem?

For whatever reason, a previous bootstrap didn't complete. This has left the stack in REVIEW_IN_PROGRESS. Running `cdk bootstrap` on the region in which the stack is stuck causes cdk to get stuck. When running without any additional options, it just hangs. It requires you to run with `--verbose --debug` to start getting some output that tells you what's going on.

### Reproduction Steps

Create a bootstrap stack and leave it in a REVIEW_IN_PROGRESS state. Then try to run cdk bootstrap

### What did you expect to happen?

It should give me an option to execute the change set. I would have thought since `--execute` is default, it would have just executed the change sets. It also shouldn't just hang. It should at least give me an error or something letting me know it's waiting. It also probably should have some kind of timeout implemented

### What actually happened?

It just hangs

### CDK CLI Version

2.15.0 (build 151055e)

### Node.js Version

v16.14.0

### OS

WSL2

### Language

Python

### Language Version

Python 3.7.13

**Comment by rix0rrr** [comment]
The initial wait for stabilization should print that it's waiting, and give up if the stack doesn't stabilize in X time. It could also say something like `Do you want to continue waiting (Y/n)?`

**Comment by chizou** [comment]
@rix0rrr that may be true, but I'm running the job in a Jenkins pipeline and Jenkins might be swallowing the output. Is there a way to always reject waiting via a cli flag?

**Comment by rix0rrr** [comment]
I see. It looks like we can treat `REVIEW_IN_PROGRESS` the same as `ROLLBACK_COMPLETE`: a stack failed to create initially, and the solution is to delete it and recreate.

The change probably needs to be made here:

https://github.com/aws/aws-cdk/blob/4f122099e2d4a6b3bdf6edfb2e99986dd266a71e/packages/aws-cdk/lib/api/util/cloudformation/stack-status.ts#L15-L18

I don't have time to change and test this myself, but contributions are welcome.

**Comment by simonespa** [comment]
I had the same problem just now and I too had to use `-v` to see what was going on.

The issue gets trickier by the fact that for some reason the CloudFormation list of stacks in the Console doesn't display the stack in `REVIEW_IN_PROGRESS` status.

How can it be in that status and not appearing? After a bit of search, I tried `aws cloudformation list-stacks` from the CLI and I noticed that the stack is indeed there.
